File: api/src/main/java/ru/dverkask/skinanatomy/utils/ImageLoader.java
package ru.dverkask.skinanatomy.utils;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.IOException;
import java.net.URL;

@Deprecated
public class ImageLoader {

    @Deprecated
    public static BufferedImage loadImage(String imageUrl) throws IOException {
        URL url = new URL(imageUrl);
        return ImageIO.read(url);
    }
}

File: api/src/main/java/ru/dverkask/skinanatomy/api/image/ImageLoader.kt
package ru.dverkask.skinanatomy.api.image

import java.io.IOException
import java.net.URL
import javax.imageio.ImageIO

object ImageLoader {
    fun loadImage(url: String): ImageLoadResult {
        return try {
            URL(url).openStream().use { stream ->
                val image = ImageIO.read(stream)
                if (image != null) {
                    ImageLoadResult.Success(image)
                } else {
                    ImageLoadResult.Error(IOException("Failed to load image: null result"))
                }
            }
        } catch (e: Exception) {
            ImageLoadResult.Error(e)
        }
    }
}